src/util/simplify_expr.cpp

```diff
@@ -2563,17 +2550,50 @@ bool simplify_exprt::simplify_node(exprt &expr)
     }
   }
   else if(expr.id()==ID_pointer_object)
-    no_change = simplify_pointer_object(expr) && no_change;
+  {
+    auto r = simplify_pointer_object(expr);
```

Comment:
How about factoring this repeated code with a helper function:
```
static bool set_if_changed(exprt &expr, const simplify_resuttt<> &result)
{
  if(result.has_changed())
    expr = result.expr;
  return result.has_changed();
}
```

Then make all these into `changed = set_if_changed(expr, simplify_pointer_object(expr)) || changed;`

Reply:
This will all collapse once all (or some majority) of the cases use the new interface!

Reply:
> This will all collapse once all (or some majority) of the cases use the new interface!

Ok.